src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,56 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+/**
+ * Deadline implements the type of Task that contains a
+ * deadline for the task to be completed by.
+ *
+ * @author Alvin Jiang Min Jun
+ * @version v0.1
+ */
+public class Deadline extends Task {
+
+    protected String by;
+    protected LocalDate date;
+
+    /**
+     * Deadline constructor that creates an instance of a Deadline object.
+     *
+     * @param description The description for the task that needs to be completed.
+     * @param by The deadline for the task to be completed by.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
```

Comment:
Perhaps 'by' could be more descriptive of its contents such as 'deadline' or 'deadlineDescription'